backend/benefit/applications/services/ahjo_integration.py

```diff
@@ -0,0 +1,213 @@
+
+        template_config = JINJA_TEMPLATES_COMPOSED[TEMPLATE_ID_COMPOSED_ACCEPTED_PUBLIC]
+        public_accepted_template = _get_template(template_config["path"])
+        file_name = template_config["file_name"].format(
+            start_number=start_number, end_number=end_number
+        )
+        html = public_accepted_template.render(
+            apps=accepted_apps,
+            year=timezone.now().year,
+            start_number=start_number,
+            end_number=end_number,
+        )
+        files.append((file_name, pdfkit.from_string(html, False), html))
+
+        template_config = JINJA_TEMPLATES_COMPOSED[
+            TEMPLATE_ID_COMPOSED_ACCEPTED_PRIVATE
+        ]
+        private_accepted_template = _get_template(template_config["path"])
+        file_name = template_config["file_name"].format(
+            start_number=start_number, end_number=end_number
+        )
+        html = private_accepted_template.render(
+            apps=accepted_apps,
+            year=timezone.now().year,
+            start_number=start_number,
+            end_number=end_number,
+        )
+        files.append((file_name, pdfkit.from_string(html, False), html))
+
+    # Rejected applications
+    rejected_apps = [app for app in apps if app.status == ApplicationStatus.REJECTED]
+    if len(rejected_apps):
+        start_number = rejected_apps[0].application_number
+        end_number = rejected_apps[-1].application_number
+
+        template_config = JINJA_TEMPLATES_COMPOSED[TEMPLATE_ID_COMPOSED_DECLINED_PUBLIC]
+        public_declined_template = _get_template(template_config["path"])
+        file_name = template_config["file_name"].format(
+            start_number=start_number, end_number=end_number
+        )
+        html = public_declined_template.render(
+            apps=rejected_apps,
+            year=timezone.now().year,
+            start_number=start_number,
+            end_number=end_number,
+        )
+        files.append((file_name, pdfkit.from_string(html, False), html))
+
+        template_config = JINJA_TEMPLATES_COMPOSED[
+            TEMPLATE_ID_COMPOSED_DECLINED_PRIVATE
+        ]
+        private_declined_template = _get_template(template_config["path"])
+        file_name = template_config["file_name"].format(
+            start_number=start_number, end_number=end_number
+        )
+        html = private_declined_template.render(
+            apps=rejected_apps,
+            year=timezone.now().year,
+            start_number=start_number,
+            end_number=end_number,
+        )
+        files.append((file_name, pdfkit.from_string(html, False), html))
+    return files
+
+
+def generate_zip(files):
+    mem_zip = BytesIO()
+
+    with zipfile.ZipFile(mem_zip, mode="w", compression=zipfile.ZIP_DEFLATED) as zf:
+        for f in files:
+            zf.writestr(f[0], f[1])
+
+    return mem_zip.getvalue()
+
+
+def export_application_batch(batch):
+    if batch.applications.count() <= 0:
```

Comment:
Handling of the application batch status: When the data is exported, application batch status should already be SENT_TO_AHJO, or alternatively the status should be set to SENT_TO_AHJO here if it is still DRAFT. The Talpa guys had requirement that the applications should be immutable after the ahjo data is generated.
Also, we need to save the id of the person doing the state transfer to SENT_TO_AHJO, that is needed in Talpa. I guess that would be automatically stored in the history table of the application batch.

Reply:
Thanks for the information, I updated the validation and tests

Reply:
Btw, I think storing the ID of person doing the state transfer should be done in a separated ticket leater, when we integrate the handler authentication API